src/nanoarrow/ipc/decoder_test.cc

```diff
@@ -984,7 +984,13 @@ TEST_P(ArrowSchemaParameterizedTestFixture, NanoarrowIpcNanoarrowFooterRoundtrip
   EXPECT_EQ(
       ArrowIpcEncoderFinalizeBuffer(encoder.get(), /*encapsulate=*/false, buffer.get()),
       NANOARROW_OK);
+
+#ifdef __BIG_ENDIAN__
+  uint32_t footer_size_le = bswap32(static_cast<int32_t>(buffer->size_bytes));
```

Comment:
I believe byteswap functions take only unsigned integers

```suggestion
  uint32_t footer_size_le = bswap32(static_cast<uint32_t>(buffer->size_bytes));
```